### PR TITLE
Fix selecting wrong schema pcid on pgpointcloud writing (#2692)

### PR DESCRIPTION
**Pull request by sensor-freak**
Also added a unit test to ensure the fix works as expected.

### CODE REVIEW

**Comment by sensor-freak** [comment]
I have pushed wrong commit. Will fix it soon.

File: plugins/pgpointcloud/io/PgWriter.cpp [hunk 1]
```diff
@@ -230,8 +230,10 @@ uint32_t PgWriter::SetupSchema(uint32_t srid)
     // Do any of the existing schemas match the one we want to use?
     if (schema_count > 0)
     {
-        PGresult *result = pg_query_result(m_session,
-            "SELECT pcid, schema FROM pointcloud_formats");
+        oss.str("");
+        oss.clear();
+        oss << "SELECT pcid, schema FROM pointcloud_formats WHERE srid = " << srid;
+        PGresult *result = pg_query_result(m_session, oss.str());
```
Comment (sensor-freak) [inline plugins/pgpointcloud/io/PgWriter.cpp]: Here I'm using oss to construct the query. oss.str("") and oss.clear() ensure that the stream is really empty.
The modified SQL query only selects those schema entries, that have the wanted SRS column. This modification will allow a new row to be generated, when the same schema is used with a different SRS. (Generation of the new entry is performed below with the INSERT statement (line 288ff).

**Comment by sensor-freak** [comment]
I have reverted some changes that shouldn't have gone to the repo at all.

**Comment by sensor-freak** [comment]
> 
> 
> > I don't understand the changes in this function. Can you explain?
> 
> The reader is not making joins properly in some situations.

I cannot comment on the reader, since I'm using only the writer.

**Comment by abellgithub** [comment]
Integrated via 59ad8229fffbb70a4f6e3b51e61d2c89bae2e654

**Pull request closed without merging**